crates/turbopack-build/src/ecmascript/node/evaluate/chunk.rs

```diff
@@ -0,0 +1,245 @@
+
+/// An Ecmascript chunk that loads a list of parallel chunks, then instantiates
+/// runtime entries.
+#[turbo_tasks::value(shared)]
+pub(crate) struct EcmascriptBuildNodeEvaluateChunk {
+    chunking_context: BuildChunkingContextVc,
+    entry_chunk: ChunkVc,
+    other_chunks: AssetsVc,
+    evaluatable_assets: EvaluatableAssetsVc,
+    exported_module: Option<EcmascriptModuleAssetVc>,
+}
+
+#[turbo_tasks::value_impl]
+impl EcmascriptBuildNodeEvaluateChunkVc {
+    /// Creates a new [`EcmascriptBuildNodeEvaluateChunkVc`].
+    #[turbo_tasks::function]
+    pub fn new(
+        chunking_context: BuildChunkingContextVc,
+        entry_chunk: ChunkVc,
+        other_chunks: AssetsVc,
+        evaluatable_assets: EvaluatableAssetsVc,
+        exported_module: Option<EcmascriptModuleAssetVc>,
+    ) -> Self {
+        EcmascriptBuildNodeEvaluateChunk {
+            chunking_context,
+            entry_chunk,
+            other_chunks,
+            evaluatable_assets,
+            exported_module,
+        }
+        .cell()
+    }
+
+    #[turbo_tasks::function]
+    async fn code(self) -> Result<CodeVc> {
+        let this = self.await?;
+
+        let output_root = this.chunking_context.output_root().await?;
+        let chunk_path = self.ident().path().await?;
+        let chunk_directory = self.ident().path().parent().await?;
+        let runtime_path = self
+            .runtime_reference()
+            .runtime_chunk()
+            .ident()
+            .path()
+            .await?;
+        let runtime_relative_path =
+            if let Some(path) = chunk_directory.get_relative_path_to(&runtime_path) {
+                path
+            } else {
+                bail!(
+                    "cannot find a relative path from the chunk ({}) to the runtime chunk ({})",
+                    chunk_path.to_string(),
+                    runtime_path.to_string(),
+                );
+            };
+        let chunk_public_path = if let Some(path) = output_root.get_path_to(&chunk_path) {
+            path
+        } else {
+            bail!(
+                "chunk path ({}) is not in output root ({})",
+                chunk_path.to_string(),
+                output_root.to_string()
+            );
+        };
+
+        let mut code = CodeBuilder::default();
+
+        writedoc!(
+            code,
+            r#"
+                const CHUNK_PUBLIC_PATH = {};
+                const runtime = require({});
+            "#,
+            StringifyJs(&*chunk_public_path),
+            StringifyJs(&*runtime_relative_path)
+        )?;
+
+        let other_chunks = this.other_chunks.await?;
+        for other_chunk in &*other_chunks {
+            let other_chunk_path = &*other_chunk.ident().path().await?;
+            if let Some(other_chunk_public_path) = output_root.get_path_to(other_chunk_path) {
+                writedoc!(
+                    code,
+                    r#"
+                        runtime.loadChunk({});
```

Comment:
```suggestion
                        runtime.loadChunk(require({}));
```

To make the bundle statically analyse-able, e. g. by nft, it would be great to avoid dynamic require() calls and inject the require call with the chunk path.

Reply:
How would that work with dynamic imports on the server-side? We transform the import into a `__turbopack_load_chunk__` call. Would that also need to be replaced with a direct call to `require` instead?

Reply:
Yeah that would be perfect

Reply:
Created https://linear.app/vercel/issue/WEB-1112/make-the-production-nodejs-runtime-statically-analyzeable to track this.